recipes/7zip/19.00/conanfile.py

```diff
@@ -33,13 +33,18 @@ def _uncompress_7z(self, filename):
         tools.get(**self.conan_data["externals"]["lzma"])
         self.run("lzma920\\7zr.exe x {}".format(filename))
 
+    _msvc_platforms = {
+        'x86_64': 'x64',
+        'x86': 'x86',
+    }
+
     def build(self):
-        if self.settings.os_build == "Windows":
+        if self.settings.compiler == "Visual Studio":
             env_build = VisualStudioBuildEnvironment(self)
             with tools.environment_append(env_build.vars):
                 vcvars = tools.vcvars_command(self.settings)
                 with tools.chdir("CPP/7zip"):
-                    self.run("%s && nmake /f makefile" % vcvars)
+                    self.run("%s && nmake /f makefile PLATFORM=%s" % (vcvars, self._msvc_platforms[str(self.settings.arch_build)]))
```

Comment:
extract function, and throw exception on unsupported platform (right now it will end up in `PLATFORM=None` which is hard to diagnose)

Reply:
I think I addressed your comment